src/org/exist/xquery/value/ValueSequence.java

```diff
@@ -793,7 +793,7 @@ public Item nextItem() {
 
     private class ReverseValueSequenceIterator implements SequenceIterator {
 
-        private int pos = size - 1;
+        private int pos = size; // size is not the actual size
```

Comment:
It's not immutable by implementation. There number of methods (like `add`, `addAll`, `clear` and etc) that change state.